src/network_models/power_flow_evaluation.jl

```diff
@@ -106,36 +134,156 @@ function _make_pf_input_map!(
+                        get_component_names(get_attributes(val))
+                    else
+                        axes(val)[1]
                     end
-                    pf_data_opt_container_map[key] = name_bus_ix_map
+                for comp_name in comp_names
+                    name_bus_ix_map[comp_name] =
+                        temp_component_map[comp_type][comp_name]
                 end
+                pf_data_opt_container_map[key] = name_bus_ix_map
             end
         end
-        pf_e_data.input_key_map[category] = pf_data_opt_container_map
+    end
+end
+
+# the function to map HVDC power transfers as bus injections is not applicable to PSSEExporter:
+_add_two_terminal_elements_map!(
+    ::PSY.System,
+    ::PFS.PSSEExporter,
+    ::Vector{Pair{OptimizationContainerKey, Any}},
+    ::Dict{Symbol, <:Dict{OptimizationContainerKey, <:Dict}},
+) = nothing
+
+"""
+    _add_two_terminal_elements_map!(
+        sys::PSY.System,
+        pf_data::PFS.PowerFlowData,
+        available_keys::Vector{Pair{OptimizationContainerKey, Any}},
+        input_key_map::Dict{Symbol, Dict{OptimizationContainerKey, Dict{String, Int64}}}
+    )
+
+Adds mappings for two-terminal elements (HVDC components) that connect the power flow results (from -> to, to -> from) 
+to be added to the mappings for all component types.
+The results for these elements are added as bus injections in the `PowerFlowData` as a simplified representation of 
+these components.
+
+# Arguments
+- `sys::PSY.System`: `System` instance representing the power system model.
+- `pf_data::PFS.PowerFlowData`: The power flow data used internally for power flow calculations.
+- `available_keys::Vector{Pair{OptimizationContainerKey, Any}}`: A vector of available optimization container keys and their associated values.
+- `input_key_map::Dict{Symbol, Dict{OptimizationContainerKey, Dict{String, Int64}}}`: A dictionary mapping categories to optimization container keys and their associated mappings. To be extended in this function by the mappings for the two-terminal elements to the respective buses in the `PowerFlowData` instance.
+"""
+function _add_two_terminal_elements_map!(
+    sys::PSY.System,
+    pf_data::PFS.PowerFlowData,
+    available_keys::Vector{Pair{OptimizationContainerKey, Any}},
+    input_key_map::Dict{Symbol, <:Dict{OptimizationContainerKey, <:Dict}},
+)
+    for (element_type, category, get_bus_func) in zip(
+        [
+            PSY.TwoTerminalHVDC,
+            PSY.TwoTerminalHVDC,
+            PSY.PhaseShiftingTransformer,
+            PSY.PhaseShiftingTransformer,
+        ],
+        [
+            :active_power_hvdc_from_to,
+            :active_power_hvdc_to_from,
+            :active_power_pst_from_to,
+            :active_power_pst_to_from,
+        ],
+        [
+            x -> PSY.get_from(PSY.get_arc(x)),
+            x -> PSY.get_to(PSY.get_arc(x)),
+            x -> PSY.get_from(PSY.get_arc(x)),
+            x -> PSY.get_to(PSY.get_arc(x)),
```

Comment:
I don't like this pattern of using anonymous functions like this. We wrote the getters in PSY to get most of that code pre-compiled and using anonymous functions defeats the purpose.

Reply:
This was needed to avoid code duplications. I will look for a different solution

Reply:
I will define them as named functions

Reply:
Do we want to define get_from and get_to to work on branches and not just arcs? So that we do not need to chain get_from(get_arc(branch)) every time?

Reply:
we could add to PSY functions for ACBranches `get_from_bus`, `get_to_bus` or something like that.